src/components/CreateMonster.vue

```diff
@@ -31,13 +32,36 @@ export default {
     }
   },
   methods: {
+    // Add monster to firebase, but only if it's less than 46 characters and doesn't already exist.
     addMonster () {
-      this.$firebaseRefs.monsters.push(this.newMonster)
-      this.$router.replace('hello')
+      // this.$firebaseRefs.monsters is how we access the monsters database (using vuefire)
+      // .once('value') listens for exactly one 'value' event from firebase
+      this.$firebaseRefs.monsters.once('value', s => {
+        // Retrieve all existing database entries into an array called 'monsterVals'
+        const monsterVals = Object.values(s.val())
+        // Put only the 'name' value of the database entries into an array called 'monsterNames'
+        const monsterNames = monsterVals.map(m => m.name)
+        // Check the length of the NEW monster's name
+        if (this.newMonster.name.length > 45) {
```

Comment:
We know the length of the name much earlier than this, so if the name is too long we've now made a network call that we could have avoided. Let's move this check to the beginning of this function.